### ISSUE TITLE
[AnalyticsAPI] Add Video Progress and Video seek tag generation using the ibmstats.event function

### ISSUE THREAD

**Issue by matejzorkoci** [carbon-design-system/carbon-for-ibm-dotcom#3716]
Hi,
we would like to capture two additional event,s when someone is watching a video on IBM page, to determine how much of that video was viewed and if they are skipping to another section of the video. 

Video Progress:
Currently, we do capture the exact time, when someone does some action on the video (play, pause, ...), but we need to have an automated event, that will be created each time a user views specific % of the video. The event should be executed when someone crosses the 25, 50, 75, 100 % mark of the video. 
Video Seek:
Generate an unique event, based on which we would know that a user has skipped a portion of the video to another section/time. In the old script, this kind of event is generated, but it is not fully definned (it uses number 9999). In the new Carbon template, I can see this event is no longer generated, only the Video play event is generated, which does not exactly define the action of the user.

### The problem
Video Progress:
We do not have a clean way how to group and categorise the users based on how much % of the video they viewed. Using this data, we will be able to do better analysis, create more advanced reports and help our page owner understand the behaviour of the users on their pages. This is an industry standard, which we currently do not have in IBM.
Video Seek;
We do not have information if someone is skipping the video, or, this information is collected with another event, making the data for that event incorrect.

### The solution
Video Progress:
Create an automated event, that would execute when a user views 25, 50, 75, 100 % of the video. This event would automatically executed, once the mark is achieved, so no user action is required to generate this event.
Video Seek:
When someone skips the video to another time, a unique event shoudl be created, that would tell us that this is the action the user did. We shoudl not have there other events, like Video Play or Video Pause, as these represent different user actions.

### Additional information
Video Progress and Video Seek: 
This should be part of the standard Video script, where all the other logic for executing the ibmstats.event tags is set.

**Comment by tweenn** [comment]
Hey @jeffchew

As far as I know by reading northstar's code, all 3 players from that framework (Youtube, IBM Video and Kaltura) follow this state/codes - that are mainly Youtube-API-Based:

```
0 -> 'launched'
1 -> 'paused'
2 -> 'played'
3 -> 'ended'
99 -> 'error'
101 -> 'cta-clk'
```
Where `101` is a "secret" event code for CTA Clicks in Mediacenter's Kaltura Player.

With that said, I'm not sure to what `9999` would be tied to, but normally super high values are tied to errors, probably Juan Leon would be the best one to tell us about it.

And I'm pretty sure there is no tracking for any of the mentioned events in neither frameworks:
* Each second
* Seek action (dragging the scrubber or pressing left and right)
* Passing the % landmarks

-------
Now, one thing I'm sure is that all of those metrics are gathered in Mediacenter for any admin that has elevated access to the media (either audio or video)

They key point to those metrics in mediacenter is that since they're "server" metrics, they can't be tied to a particular anonimized user or any campaign, etc., so having then in GA would be awesome.

-------

TL DR:
I can add those trackers and logic into the kaltura-service under the service package - as part of https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7488, but I'll need a hand from Juan's team to know which state-code to use and what extra data the seek-action will probably need - like telling from which second it jumped from.
